repository_service_tuf_worker/repository.py

```diff
@@ -607,108 +679,383 @@ def save_settings(self, root: Metadata[Root], roles_info: Dict[str, Any]):
     ):
         """Setup target delegations no matter if succinct hash bin or custom"""
         delegated_roles: List[str] = []
-        if custom_targets:
-            # Using custom Target roles delegations with path prefixes.
-            targets.signed.delegations = Delegations({}, {})
-            for role_name, role_info in custom_targets.items():
-                keyid = online_pub_key.keyid
-                targets.signed.delegations.roles[role_name] = DelegatedRole(
-                    role_name, [keyid], 1, True, role_info["path_patterns"]
+
+        # Using succinct hash bin delegations.
+        # Calculate the bit length (Number of bits between 1 and 32)
+        # Calculate the bit length (Number of bits between 1 and 32)
+        bit_length = int(
+            log(self._settings.get_fresh("NUMBER_OF_DELEGATED_BINS"), 2)
+        )
+        # Succinct delegated roles (`bins`)
+        succinct_roles = SuccinctRoles([], 1, bit_length, BINS)
+        targets.signed.delegations = Delegations(
+            keys={}, succinct_roles=succinct_roles
+        )
+        # Initialize all succinct delegated roles (`bins`), update expire,
+        # sign, add to `Snapshot` meta and persist in the backend storage
+        # service.
+        db_target_roles: List[targets_schema.RSTUFTargetRoleCreate] = []
+
+        for delegated_name in succinct_roles.get_roles():
+            targets.signed.add_key(self._online_key, delegated_name)
+            bins_role = Metadata(Targets())
+            db_target_roles.append(
+                targets_schema.RSTUFTargetRoleCreate(
+                    rolename=delegated_name, version=1
                 )
-                targets.signed.add_key(online_pub_key, role_name)
-                custom_target = Metadata(Targets())
-                self._bump_expiry(custom_target, role_name)
-                self._sign(custom_target)
-                self._persist(custom_target, role_name)
-                delegated_roles.append(role_name)
+            )
+            self._bump_expiry(bins_role, BINS)
+            self._sign(bins_role)
+            self._persist(bins_role, delegated_name)
+            delegated_roles.append(delegated_name)
+
+        targets_crud.create_roles(self._db, db_target_roles)
+
+    def _remove_delegated_role_keys(
+        self, targets: Metadata[Targets], delegated: DelegatedRole
+    ):
+        other_keys = []
+        logging.debug("mapping keys")
+        for not_delete in targets.signed.delegations.roles:
+            if not_delete == delegated:
+                continue
+            other_keys += targets.signed.delegations.roles[not_delete].keyids
+
+        role_keys = targets.signed.delegations.roles[delegated.name].keyids
+        for key in role_keys:
+            if key not in other_keys:
+                logging.debug(f"removing key id {key}")
+                targets.signed.delegations.keys.delete(key)
+            logging.debug(f"key {key} used by other role")
```

Comment:
This log should be in an else otherwise it's confusing.
```suggestion
            if key not in other_keys:
                logging.debug(f"removing key id {key}")
                targets.signed.delegations.keys.delete(key)
             else:
                logging.debug(f"key {key} used by other role")
```